### PR TITLE
refactor: Use less plugin

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #20365

### CODE REVIEW

File: components/button/style/mixin.less [hunk 1]
```diff
@@ -33,28 +35,20 @@
   &:hover,
   &:focus {
     & when (@theme = dark) {
-      .button-color(
-        @color; ~`colorPalette('@{background}', 7) `; ~`colorPalette('@{background}', 7) `
-      );
+      .button-color(@color; color-palette(@background, 7) ; color-palette(@background, 7));
```
Comment (ycjcl868) [inline components/button/style/mixin.less]:
```suggestion
      .button-color(@color; colorPalette(@background, 7) ; colorPalette(@background, 7));
```

Comment (zombieJ) [reply components/button/style/mixin.less]: plugin 命名按照 less function 的命名风格来。和 js function 区分一下。

File: components/style/color/color-palette.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+const tinyColor = require('./tiny-color');
```
Comment (ycjcl868) [inline components/style/color/color-palette.js]: 这个 js 可以直接用 `@ant-design/colors` 的 `generate` 函数吗？

File: components/style/themes/dark.less [hunk 3]
```diff
@@ -2,161 +2,161 @@
 
 @theme: dark;
 // color palettes
-@blue-1: mix(color(~`colorPalette('@{blue-base}', 8) `), @component-background, 15%);
```
Comment (ycjcl868) [inline components/style/themes/dark.less]: 用户也需要安装这个 `color-palette` less 插件才可以换暗色主题吗？

File: components/style/color/color-palette.js (hunk 2 above)
Comment (zombieJ) [reply components/style/color/color-palette.js]: less 使用 的自己的解析器，可以把 tiny-color 的代码弄到 antd colors 里。直接用 colors 的代码 less 解析不了。

Comment (ycjcl868) [reply components/style/color/color-palette.js]: 计算逻辑可以共用一套吧，tinycolor 这个

**Comment by zombieJ** [comment]
Seems OOM when dist bundle.

**Comment by zombieJ** [comment]
Hi @matthew-dean, seems OOM with plugin. Is there any clue for tracking this?

`npm run dist`

**Comment by zombieJ** [comment]
`antd.css` is not correct after adjust:

截屏2019-12-26下午10 15 11

**Comment by zombieJ** [comment]
Tested seems only variable naming selector will missing.

**Comment by PengJiyuan** [comment]
@matthew-dean

**Pull request closed without merging**